File: mobile/src/main/java/at/shockbytes/remote/network/message/MessageSerializer.java
package at.shockbytes.remote.network.message;

/**
 * @author Martin Macheiner
 *         Date: 27.09.2017.
 */

public interface MessageSerializer {

    String mouseMoveMessage(int deltaX, int deltaY);

    String writeTextMessage(int keyCode, boolean isUpperCase);

    String keyExchange(String deviceName, String certificate);
}

File: mobile/src/main/java/at/shockbytes/remote/network/message/MessageSerializer.kt
package at.shockbytes.remote.network.message

/**
 * @author Martin Macheiner
 * Date: 27.09.2017.
 */

interface MessageSerializer {

    fun mouseMoveMessage(deltaX: Int, deltaY: Int): String

    fun writeTextMessage(keyCode: Int, isUpperCase: Boolean): String

    fun keyExchange(deviceName: String, certificate: String): String
}
